flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/rules/logical/LogicalWindowAggregateRuleBase.scala

```diff
@@ -88,6 +85,9 @@ abstract class LogicalWindowAggregateRuleBase(description: String)
       .project(project.getChildExps.updated(windowExprIdx, inAggGroupExpression))
       .build()
 
+    // translate window against newProject.
+    val window = translateWindow(windowExpr, windowExprIdx, newProject.getRowType)
```

Comment:
Because `translateWindow` is postponed after `getInAggregateGroupExpression`, so if the time field is not a time attribute, the exception will be thrown from `getInAggregateGroupExpression`. But the exception there is not clear enought. Could you improve the exception in `getInAggregateGroupExpression`? For example: 

```scala
s"Window aggregate can only be defined over a time attribute column, but ${timeAttribute.getType} encountered."
```